### PR TITLE
Add Material 3 `ProgressIndicator` examples

### PR DESCRIPTION
**Pull request by TahaTesser**
part of https://github.com/flutter/flutter/issues/101048

### M3 `CircularProgressIndicator` example

https://user-images.githubusercontent.com/48603081/197593820-4454b109-f01e-488b-bbc4-e6b15417fadf.mov

### M3 `LinearProgressIndicator` example

https://user-images.githubusercontent.com/48603081/197594151-c650695b-a4fd-4df3-8409-a253892fe1e7.mov

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I signed the [CLA].
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

File: examples/api/lib/material/progress_indicator/circular_progress_indicator.1.dart [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+}
+
+class ProgressIndicatorExample extends StatefulWidget {
+  const ProgressIndicatorExample({super.key});
+
+  @override
+  State<ProgressIndicatorExample> createState() => _ProgressIndicatorExampleState();
+}
+
+/// [AnimationController]s can be created with `vsync: this` because of
```
Comment (gspencergoog) [inline examples/api/lib/material/progress_indicator/circular_progress_indicator.1.dart:30]: Maybe move this comment down to just above the `vsync:` line?

File: examples/api/lib/material/progress_indicator/linear_progress_indicator.0.dart [hunk 2]
```diff
@@ -6,32 +6,29 @@
...
-  const MyStatefulWidget({super.key});
+class ProgressIndicatorExample extends StatefulWidget {
+  const ProgressIndicatorExample({super.key});
 
   @override
-  State<MyStatefulWidget> createState() => _MyStatefulWidgetState();
+  State<ProgressIndicatorExample> createState() => _ProgressIndicatorExampleState();
 }
 
 /// [AnimationController]s can be created with `vsync: this` because of
```
Comment (gspencergoog) [inline examples/api/lib/material/progress_indicator/linear_progress_indicator.0.dart:29]: Same here.

File: examples/api/lib/material/progress_indicator/linear_progress_indicator.1.dart [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+}
+
+class ProgressIndicatorExample extends StatefulWidget {
+  const ProgressIndicatorExample({super.key});
+
+  @override
+  State<ProgressIndicatorExample> createState() => _ProgressIndicatorExampleState();
+}
+
+/// [AnimationController]s can be created with `vsync: this` because of
```
Comment (gspencergoog) [inline examples/api/lib/material/progress_indicator/linear_progress_indicator.1.dart:30]: And here

File: packages/flutter/lib/src/material/progress_indicator.dart [hunk 4]
```diff
@@ -474,6 +482,14 @@ class _CircularProgressIndicatorPainter extends CustomPainter {
 /// ** See code in examples/api/lib/material/progress_indicator/circular_progress_indicator.0.dart **
 /// {@end-tool}
 ///
+/// {@tool dartpad}
+/// This sample shows the creation of a [CircularProgressIndicator] with a changing value.
+/// When toggling the switch, [CircularProgressIndicator] uses a determinate value.
+/// As described in: https://m3.material.io/components/progress-indicators/overview
+///
+/// ** See code in examples/api/lib/material/progress_indicator/linear_progress_indicator.1.dart **
```
Comment (gspencergoog) [inline packages/flutter/lib/src/material/progress_indicator.dart:490]: Is this the right filename? Seems like it should be circular_progress_indicator.1.dart.

File: examples/api/lib/material/progress_indicator/circular_progress_indicator.1.dart (hunk 1 above)
Comment (TahaTesser) [reply examples/api/lib/material/progress_indicator/circular_progress_indicator.1.dart]: Sure!

File: packages/flutter/lib/src/material/progress_indicator.dart (hunk 4 above)
Comment (TahaTesser) [reply packages/flutter/lib/src/material/progress_indicator.dart]: My bad. Fixed.

File: examples/api/test/material/progress_indicator/linear_progress_indicator.0.dart [hunk 5]
```diff
@@ -0,0 +1,24 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
```
Comment (gspencergoog) [inline examples/api/test/material/progress_indicator/linear_progress_indicator.0.dart:1]: Shouldn't the name of this file be: `examples/api/test/material/progress_indicator/linear_progress_indicator.0_test.dart`?

File: examples/api/test/material/progress_indicator/linear_progress_indicator.1.dart [hunk 6]
```diff
@@ -0,0 +1,36 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
```
Comment (gspencergoog) [inline examples/api/test/material/progress_indicator/linear_progress_indicator.1.dart:1]: Same here, this filename should end in `_test.dart`

File: examples/api/test/material/progress_indicator/linear_progress_indicator.0.dart (hunk 5 above)
Comment (TahaTesser) [reply examples/api/test/material/progress_indicator/linear_progress_indicator.0.dart]: Fixed :)

**Review by gspencergoog — APPROVED** [review]
32384589-a60f0e74-c078-11e7-9bc1-e5b5287aea9d

**Pull request merged**